### PR TITLE
Ensure static certs in kubeconfig override exec plugin

### LINKED ISSUE #99603: Exec plugin overrides static certs in kubeconfig (different behavior than over statically configured credentials)
#### What happened:

- My exec plugin overrode the static certs I had in my kubeconfig.

#### What you expected to happen:

- The static certs I had in my kubeconfig should take precedent over whatever is configured by my exec plugin.

#### How to reproduce it (as minimally and precisely as possible):

With admin-like privileges, do the following.
```bash
# Generate CSR.
openssl req -nodes -subj /CN=some-username -newkey rsa:2048 -keyout /tmp/aaa.key -out /tmp/aaa.csr

# Create CSR.
cat <<EOF | kubectl replace -f -
apiVersion: certificates.k8s.io/v1
kind: CertificateSigningRequest
metadata:
  name: aaa
spec:
  request: $(cat /tmp/aaa.csr | base64 | tr -d '\n')
  signerName: kubernetes.io/kube-apiserver-client
  usages: [client auth]
EOF

# Issue certificate.
kubectl certificate approve aaa
kubectl get csr aaa -o jsonpath={.status.certificate} | base64 -d > /tmp/aaa.crt

# Make request with both certificate and exec plugin that returns token.
cat <<EOF >/tmp/aaa.sh
#!/bin/bash
echo '{"kind":"ExecCredential","apiVersion":"client.authentication.k8s.io/v1beta1","status":{"clientKeyData":"bad","clientCertificateData":"bad"}}'
EOF
chmod +x /tmp/aaa.sh
kubectl config set-credentials aaa --client-certificate=/tmp/aaa.crt --client-key=/tmp/aaa.key --exec-api-version client.authentication.k8s.io/v1beta1 --exec-command=/tmp/aaa.sh
kubectl --user aaa get pods
```

#### Anything else we need to know?:

- This seems like unexpected behavior because other statically configured credentials (i.e., basic auth, bearer token) take precedent over the stuff returned by the exec plugin.
- This was pointed out by @enj when he was reviewing some exec plugin integration tests.

#### Environment:
- Kubernetes version (use `kubectl version`): 1.20.0
- Cloud provider or hardware configuration: `kind`, but shouldn't matter
- OS (e.g: `cat /etc/os-release`): Darwin, but shouldn't matter
- Kernel (e.g. `uname -a`): Darwin, but shouldn't matter
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

### PR DESCRIPTION
**Pull request by margocrawf**
Check whether static cert is already configured in UpdateTransportConfig
- Also update test-cmd.sh to pass a signing ca to the kube controller
  manager, so CSRs work properly in integration tests.

fixed #99603

/kind bug

```release-note
Fixed a bug that caused credentials in an exec plugin to override the static certificates set in a kubeconfig.
```

### CODE REVIEW

**Comment by neolit123** [comment]
> This seems like unexpected behavior because other statically configured credentials (i.e., basic auth, bearer token) take precedent over the stuff returned by the exec plugin.

Dynamic (plugins) taking precedence over static config seems like the more intuative pattern to me. But i am not very familiar with area of the code and it seems that making this alternative switch now can be more breaking.

File: hack/testdata/auth/testcert.csr [hunk 1]
```diff
@@ -0,0 +1,15 @@
+-----BEGIN CERTIFICATE REQUEST-----
```
Comment (enj) [inline hack/testdata/auth/testcert.csr:1]: For all of these new testdata files, add:

1. A comment describing how they were generated 
2. A note saying that this is a test-only credential and that no security vulnerabilities should be reported in regards to them being published to a public repo

File: test/cmd/authentication.sh [hunk 2]
```diff
@@ -134,6 +134,50 @@ EOF
...
+  ### Provided --client-certificate/--client-key should take precedence on the cli, thus not triggering the (invalid) exec credential plugin
+
+  # Use CSR so we have a valid cert pair
+  cat <<EOF | kubectl create -f -
+apiVersion: certificates.k8s.io/v1
+kind: CertificateSigningRequest
+metadata:
+  name: testcert
+spec:
+  request: $(base64 < hack/testdata/auth/testcert.csr | tr -d '\n')
```
Comment (enj) [inline test/cmd/authentication.sh:146]: It is unclear what `testcert.csr` encodes.

**Review by enj — CHANGES_REQUESTED** [review]

**Pull request merged**